libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -192,11 +223,118 @@ bool AP_OABendyRuler::update(const Location& current_loc, const Location& destin
     destination_new.offset_bearing(chosen_bearing, distance_to_dest);
 
     // log results
-    AP::logger().Write_OABendyRuler(true, chosen_bearing, false, best_margin, destination, destination_new);
+    AP::logger().Write_OABendyRuler((uint8_t)OABendyType::OA_BENDY_HORIZONTAL, true, chosen_bearing, 0.0f, false, best_margin, destination, destination_new);
+
+    return true;
+}
+
+// Search for path in the Vertical directions
+bool AP_OABendyRuler::search_vertical_path(const Location& current_loc, const Location& destination,Location &destination_new, const float &lookahead_step1_dist, const float &lookahead_step2_dist, const float &bearing_to_dest, const float &distance_to_dest) 
+{
+    // check OA_BEARING_INC_VERTICAL definition allows checking in all directions
+    static_assert(360 % OA_BENDYRULER_BEARING_INC_VERTICAL == 0, "check 360 is a multiple of OA_BEARING_INC_VERTICAL");
+    float best_vertical_bearing = 0.0f;
+    bool  have_best_vertical_bearing = false;
+    float best_margin = -FLT_MAX;
+    float best_margin_pitch = best_vertical_bearing;
+    const uint8_t limit = 180 / OA_BENDYRULER_BEARING_INC_VERTICAL;
+
+    for (uint8_t i = 0; i <= limit; i++) {
+        for (uint8_t bdir = 0; bdir <= 1; bdir++) {
+            // skip duplicate check of bearing straight towards destination or 180 degrees behind
+            if (((i==0) && (bdir > 0)) || ((i == limit) && (bdir > 0))) {
+                continue;
+            }
+
+            // bearing that we are probing
+            const float vertical_bearing_delta = i * OA_BENDYRULER_BEARING_INC_VERTICAL * (bdir == 0 ? 1.0f : -1.0f);
+
+            Location test_loc = current_loc;
+            test_loc.offset_bearing_and_pitch(bearing_to_dest, vertical_bearing_delta,lookahead_step1_dist);
+
+            // calculate margin from fence for this scenario
+            float margin = calc_avoidance_margin(current_loc, test_loc);
+
+            if (margin > best_margin) {
+                best_margin_pitch = vertical_bearing_delta;
+                best_margin = margin;
+            }
+
+            if (margin > _margin_max) {
+                //This path avoids the obstacles with the required margin, now check for the path ahead
+                if (!have_best_vertical_bearing) {
+                    best_vertical_bearing = vertical_bearing_delta;
+                    have_best_vertical_bearing = true;
+                }
+                const float test_vertical_bearings_step2[] { 0.0f, 90.0f, -90.0f, 180.0f};
+                float bearing_to_dest2;
+                is_equal(fabsf(vertical_bearing_delta), 90.0f) ? (bearing_to_dest2 = bearing_to_dest) : (bearing_to_dest2 = test_loc.get_bearing_to(destination) * 0.01f);
```

Comment:
We don't normally use this pattern, instead probably more consistent to just use an if then else.